src/systems/filecoin_vm/runtime/impl/runtime.go

```diff
@@ -508,13 +491,15 @@ func (rt *VMContext) Randomness(e block.ChainEpoch, offset uint64) util.Randomne
 }
 
 func (rt *VMContext) NewActorAddress() addr.Address {
-	seed := &ActorExecAddressSeed_I{
-		creator_:            rt._immediateCaller,
-		toplevelCallSeqNum_: rt._toplevelSenderCallSeqNum,
-		internalCallSeqNum_: rt._internalCallSeqNum,
-	}
-	hash := addr.ActorExecHash(Serialize_ActorExecAddressSeed(seed))
-
+	buf := new(bytes.Buffer)
+	_, err := buf.Write(addr.Serialize_Address(rt._immediateCaller))
+	util.Assert(err != nil)
+	err = binary.Write(buf, binary.BigEndian, rt._toplevelSenderCallSeqNum)
+	util.Assert(err != nil)
+	err = binary.Write(buf, binary.BigEndian, rt._internalCallSeqNum)
+	util.Assert(err != nil)
```

Comment:
why the shift from the hash of the serialized seed object to the buffer? I don't mind it. Just wondering. 

@jzimmerman please confirm there's no possible collision here. I don't see any given fixed input sizes

Reply:
Thanks, I should have called this semantic change out explicitly.

The ActorExecAddressSeed type is unnecessary, we can very simply define the output here without it. I copied the implementation from Lotus.

Reply:
SGTM